tests/integration_test/shadow_system_test.c

```diff
@@ -0,0 +1,932 @@
+ */
+static bool receivedDeleteAcceptedResult = false;
+
+/**
+ * @brief Flag to represent result from a /delete/rejected sending from the broker.
+ */
+static bool receivedDeleteRejectedResult = false;
+
+/**
+ * @brief Flag to represent result from a /get/accepted is received from the broker.
+ */
+static bool receivedGetAcceptedResult = false;
+
+/**
+ * @brief Flag to represent result from a /get/rejected  is received from the broker.
+ */
+static bool receivedGetRejectedResult = false;
+
+/**
+ * @brief Global entry time to use as a reference timestamp in the getTimeMs()
+ * function. getTimeMs() function will always return the difference between the
+ * current time and the global entry time. This reduces the chances of overflow
+ * for the 32 bit unsigned integer used for holding the timestamp.
+ */
+static uint32_t globalEntryTime;
+
+/**
+ * @brief The timer query function provided to the MQTT context.
+ *
+ * @return Time in milliseconds.
+ */
+static uint32_t getTimeMs();
+
+/**
+ * @brief Sends an MQTT CONNECT packet over the already connected TCP socket.
+ *
+ * @param[in] pContext MQTT context pointer.
+ * @param[in] pNetworkContext Network context for OpenSSL transport implementation.
+ * @param[in] createCleanSession Creates a new MQTT session if true.
+ * If false, tries to establish the existing session if there was session
+ * already present in broker.
+ * @param[out] pSessionPresent Session was already present in the broker or not.
+ * Session present response is obtained from the CONNACK from broker.
+ */
+static void establishMqttSession( MQTTContext_t * pContext,
+                                  NetworkContext_t * pNetworkContext,
+                                  bool createCleanSession,
+                                  bool * pSessionPresent );
+
+/**
+ * @brief The application callback function that is expected to be invoked by the
+ * MQTT library for incoming publish and incoming acks received over the network.
+ *
+ * @param[in] pContext MQTT context pointer.
+ * @param[in] pPacketInfo Packet Info pointer for the incoming packet.
+ * @param[in] pDeserializedInfo Deserialized information from the incoming packet.
+ */
+static void eventCallback( MQTTContext_t * pContext,
+                           MQTTPacketInfo_t * pPacketInfo,
+                           MQTTDeserializedInfo_t * pDeserializedInfo );
+
+/*-----------------------------------------------------------*/
+
+static uint32_t getTimeMs()
+{
+    TickType_t tickCount = 0;
+    uint32_t timeMs = 0UL;
+
+    /* Get the current tick count. */
+    tickCount = xTaskGetTickCount();
+
+    /* Convert the ticks to milliseconds. */
+    timeMs = ( uint32_t ) tickCount * MILLISECONDS_PER_TICK;
+
+    /* Reduce globalEntryTime from obtained time so as to always return the
+     * elapsed time in the application. */
+    timeMs = ( uint32_t ) ( timeMs - globalEntryTime );
+
+    return timeMs;
+}
+
```

Comment:
Does running uncrustify not add a function separator after each function? 

```
/*-----------------------------------------------------------*/
```

Reply:
No, it doesn't add the function separator.